test/Parser.unittest.js

```diff
@@ -246,6 +246,50 @@ describe("Parser", () => {
 				abc: ["xyz"],
 				fgh: ["xyz"]
 			}
+		],
+		"literal var definition": [
+			function() {
+				var smu = "cbd";
+			},
+			{
+				smu: "Literal"
+			}
+		],
+		"null var definition ": [
+			function() {
+				var smu;
+				smu = 123;
+			},
+			{
+				smu: null
+			}
+		],
+		"object var definition": [
+			function() {
+				var smu = {
+					smu: 123,
+					bla: 3456
+				};
+			},
+			{
+				smu: "ObjectExpression"
+			}
+		],
+		"array var decalaration": [
+			function() {
+				var smu = [1, 2, 3];
+			},
+			{
+				smu: "ArrayExpression"
+			}
+		],
+		"array destruct var decalaration": [
+			function() {
+				var [{ smu }] = [{ testVar: "abc" }, { otherVar: "bcd" }];
+			},
+			{
+				smu: "ArrayExpression"
```

Comment:
This is not correct. It should be null or undefined.

Reply:
that's what came out of `declarator.init`, could you please elaborate why so I can try and figure out what went wrong?

Sorry about the delay, I've been caught up in so many things this summer.

Reply:
`var [{ smu }] = [{ testVar: "abc" }, { otherVar: "bcd" }];`

This destructs the array and object resulting in `smu` being undefined and not an `ArrayExpression`. You need to do the same destruction when determining the initExpresssion for a variable.